src/core/component.test.ts

```diff
@@ -91,6 +103,67 @@ describe("option processing", () => {
     });
 });
 
+describe("nested options", () => {
+
+    it("pulls initital values on mounting (render function + createElement)", () => {
+        const widget = new TestComponent({
```

Comment:
Can you rewrite these tests using the template option?

Reply:
I think it's better to keep the first two tests (as they check different ways of mounting) and add one more test for the `template` option.

As the `template` option is more readable, I'm going to rewrite the last test to use it.

So, my point is to keep the mounting tests as is and use `template`s in others.